kite-data/kite-data-core/src/test/java/org/kitesdk/data/spi/TestRefinableViews.java

```diff
@@ -819,4 +820,27 @@ public void testNotPartitioned() throws Exception {
     Assert.assertNotNull("with should succeed",
         notPartitioned.with("timestamp", now));
   }
+
+  @Test
+  public void testReadySignals() {
+    Assume.assumeTrue(unbounded instanceof ReadiableView);
+    final long instant = new DateTime(2013, 10, 1, 0, 0, DateTimeZone.UTC).getMillis();
+    final ReadiableView withSpecificTimestamp =
+        (ReadiableView) unbounded.with("timestamp", instant);
+
+    Assert.assertFalse("Should not be ready initially", withSpecificTimestamp.isReady());
+
+    withSpecificTimestamp.signalReady();
+
+    Assert.assertTrue("Should be ready after signal", withSpecificTimestamp.isReady());
+
+    final ReadiableView beforeSpecificTimestamp =
+        (ReadiableView) unbounded.toBefore("timestamp", instant);
+    Assert.assertFalse("Other views should not be ready", beforeSpecificTimestamp.isReady());
+    final ReadiableView immediatelyAfterSpecificTimestamp =
+        (ReadiableView) unbounded.with("timestamp", instant+1);
+    Assert.assertFalse("Views in the same partition should not be ready",
+        immediatelyAfterSpecificTimestamp.isReady());
```

Comment:
Along with the changes to include a more exhaustive set of tests around the different ways to specify the view constraints I updated this to be more of boundary test around comparable values. Covering instants immediately before and after, containing ranges, and boundaries before and after the ready view (inclusive/exclusive)

a794050